resources/idlharness.js

```diff
@@ -3288,15 +3292,21 @@ function idl_test(srcs, deps, idl_setup_func) {
         deps = (deps instanceof Array) ? deps : [deps] || [];
         var setup_error = null;
         return Promise.all(
-            srcs.concat(deps).map(function(spec) {
-                return fetch_spec(spec);
-            }))
-            .then(function(idls) {
+            srcs.concat(deps).map(fetch_spec))
+            .then(function(results) {
+                const astArray = results.map(result =>
+                    WebIDL2.parse(result.idl, { sourceName: result.spec })
+                );
+                const validations = WebIDL2.validate(astArray);
+                if (validations.length) {
+                    const message = validations.map(v => v.message).join("\n\n");
+                    setup_error = setup_error || new Error(message);
+                }
                 for (var i = 0; i < srcs.length; i++) {
-                    idl_array.add_idls(idls[i]);
+                    idl_array.internal_add_idls(astArray[i], options);
```

Comment:
Whoops. `options` is available in `add_dependency_idls`, but dropped between then and actually executing `idl_test`, let's revert.
```suggestion
                    idl_array.internal_add_idls(astArray[i]);
```

Reply:
Oops, should have double checked that.